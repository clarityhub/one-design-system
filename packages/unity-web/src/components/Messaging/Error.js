import React from 'react';
import { any, oneOfType, bool, string } from 'prop-types';

import Box from '../../scaffolding/Box';
import Typography from '../Typography';
import Notification from '../Notification';
import Button, { ButtonSet } from '../Buttons';

import errorToString from './errorToString';

const handleRefresh = (refresh) => (e) => {
	e.preventDefault();

	if (typeof refresh === 'string') {
		window.location.href = refresh;
		return;
	}

	window.location.reload();
};

const FloatingError = ({ error, refresh = false, ...props }) => {
	return (
		<Notification type="danger" variant="block" {...props}>
			<Typography color="white">
				{errorToString(error)}
			</Typography>

			{refresh && (
				<Box margin={{ top: 'small' }}>
					<ButtonSet>
						<Button type="white" onClick={handleRefresh(refresh)}>
							Refresh
						</Button>
					</ButtonSet>

				</Box>
			)}
		</Notification>
	);
};

FloatingError.propTypes = {
	error: any,
	refresh: oneOfType([bool, string]),
};

export default FloatingError;