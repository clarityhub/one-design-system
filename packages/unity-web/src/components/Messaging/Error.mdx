---
name: Error
menu: Components
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import { mdiNotebookOutline } from '@mdi/js';
import Error from './Error';
import Button from '../Buttons';

# Error

Show an Error message when there is an `Error` to display. You can optionally
provide a refresh or redirect link to the user.

## \<Error /\> Component

```js
import { Error } from '@clarityhub/unity-web/lib/components/Messaging';
```

## Prop Types

### Error Props

<PropsTable of={Error} />

## Examples


<Playground>
    <Error />
</Playground>


<Playground>
    <Error
        error={'Any error message shows up here'}
        refresh
    />
</Playground>
