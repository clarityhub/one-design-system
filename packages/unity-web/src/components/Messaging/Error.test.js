import React from 'react';
import renderer from 'react-test-renderer';

import Error from './Error';

describe('<Error />', () => {
	it('renders default', () => {
		const node = (
			<Error />
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('renders with message', () => {
		const node = (
			 <Error
				error={'Any error message shows up here'}
				refresh
			/>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});