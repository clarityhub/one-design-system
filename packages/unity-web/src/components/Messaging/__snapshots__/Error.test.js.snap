// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders default 1`] = `
<div
  className="css-5swc56-base-danger-block"
  role="alert"
>
  <div
    className="css-d4n4rw"
  >
    <p
      className="css-1imar2r"
    >
      Something bad happened
    </p>
  </div>
</div>
`;

exports[`<Error /> renders with message 1`] = `
<div
  className="css-5swc56-base-danger-block"
  role="alert"
>
  <div
    className="css-d4n4rw"
  >
    <p
      className="css-1imar2r"
    >
      Any error message shows up here
    </p>
    <div
      className="css-15uzpn3-StyledBox-StyledBox-StyledBox-StyledBox e1b7liwf0"
    >
      <div
        className="css-1f6tjue-ButtonSet e869cgw0"
      >
        <button
          className="-white css-1qfeqqs-Button-Button entsyi60"
          onClick={[Function]}
          type="button"
        >
          Refresh
        </button>
      </div>
    </div>
  </div>
</div>
`;
